docs/src/extend/firmware-management/building-image/rugpi/tutorial.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Building an image
+tags: [Extend, Build]
+sidebar_position: 3
+---
+
+This guide walks you through building a custom image using [Rugpi](https://github.com/silitics/rugpi). Rugpi is an open source tool to build images with support for robust Over-the-Air (OTA) updates. Currently it only supports Raspberry Pi devices, however in the future more device types may be considered.
+
+Rugpi is very beginner friendly compared to The Yocto Project, and has much shorter build times since it works by extending the official Raspberry Pi images with custom recipes which either install or configure components in the image, rather than rebuilding everything from scratch. Building an image with custom recipes only takes about 10 mins compared with the > 4 hours using The Yocto Project.
+
+The instructions on this page can be used to build images for the following devices:
+
+* Raspberry Pi 1 (32 bit)
+* Raspberry Pi 2 (64 bit)
+* Raspberry Pi 3 (64 bit)
+* Raspberry Pi 4 (64 bit)
+* Raspberry Pi 5 (64 bit)
+* Raspberry Pi Zero W (32 bit)
+* Raspberry Pi Zero 2W (64 bit)
+
+Please check out the [official Rugpi documentation](https://oss.silitics.com/rugpi/) for more details on how to use and customize it.
+
+## Setting up your build system
+
+Building images using Rugpi is considerably faster and easier when compared to The Yocto Project, so it makes it a great tool for beginners.
+
+The following tools are required to build an image:
+
+* Docker Engine
+
+:::tip
+Images can be built using Rugpi using a CI Workflow. An example for a Github Workflow is included in the template project.
+:::
+
+## Building your image
+
+The [tedge-rugpi-images](https://github.com/thin-edge/tedge-rugpi-image) is project which includes out of the box configuration to perform robust Over-the-Air Operating System updates.
+
+Feel free to clone the project if you want to make your own customizations, however please always refer back to the project if you run into any problems (as it may have changed in the meantime).
+
+
+### Cloning the project
+
+1. Clone the project and change directory to the locally checked out folder
+
+    ```sh
+    git clone https://github.com/thin-edge/tedge-rugpi-image.git
+    cd tedge-rugpi-image
+    ```
+
+2. Install [just](https://just.systems/man/en/chapter_5.html) which is used to run different tasks provided by the project
+
+    ```sh
+    curl --proto '=https' --tlsv1.2 -sSf https://just.systems/install.sh | sudo bash -s -- --to /usr/bin
+    ```
+
+    Check out the [justfile website](https://just.systems/man/en/chapter_5.html) for more installation options.
+
+### Customizing your own image
+
+1. Fork the project
+
+2. Edit the authorized ssh keys for each profile
```

Comment:
It might be good to stress that this step is critical to connect a device booting this image.

Reply:
Resolved by a929c2d124cab9c2800b80992e6011d35e25d563